ecpp/current.ecpp

```diff
@@ -222,25 +222,35 @@ std::vector<uint32_t> asset_ids;
         }
 
         std::map <std::string, double> measurements{};
+        std::map <std::string, std::string> str_measurements{}; // for values of type string
         {
-          fty::shm::shmMetrics mresult;
-          if(fty::shm::read_metrics(asset.item.name, ".*", mresult) != 0 ) {
-            log_warning ("Error reply for device '%s'", asset.item.name.c_str ());
-            continue;
-          }
-          if(mresult.size() == 0) {
-            continue;
-          }
-          for(auto &data1 : mresult) {
-              // TODO: non double values are not (yet) supported
-              double dvalue = 0.0;
-              try {
-                  dvalue = std::stod (fty_proto_value (data1));
-              } catch (const std::exception& e) {
-                  log_error ("fty_proto_value () returned a string that does not encode a double value: '%s'. Defaulting to 0.0 value.", fty_proto_value (data1));
-              }
-              measurements.emplace (std::make_pair (fty_proto_type (data1), dvalue));
-          }
+            fty::shm::shmMetrics mresult;
+            if(fty::shm::read_metrics(asset.item.name, ".*", mresult) != 0 ) {
+                log_warning ("Error reply for device '%s'", asset.item.name.c_str ());
+                continue;
+            }
+            if(mresult.size() == 0) {
+                continue;
+            }
+            for(auto &data1 : mresult) {
+                double dvalue = 0.0;
+                std::string item_name = fty_proto_type (data1);
+                // TODO: non double values are not (yet) supported
+                // Special treatment for battery status and battery charger status and battery alarm code (string values)
+                if( item_name == "status.battery" || item_name == "status.battery.charger" || item_name == "battery.alarm.code") {
```

Comment:
"operatingStatus", "status.ups" exceptions can be safely handled here using str_measurements map...

Reply:
Not possible because values are numerics